src/lib/sql/sql_translator.cpp

```diff
@@ -344,6 +347,51 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_select(const hsql::Se
   return current_result_node;
 }
 
+auto SQLTranslator::_classify_join_conditions(const hsql::JoinDefinition& join,
```

Comment:
Please, no auto-return type without unavoidable need. Pass join, left, right in as ref-params.

Reply:
Also, member functions are favoured over lambda callbacks, especially since this is always the same lambda?

Reply:
> Please, no auto-return type without unavoidable need. Pass join, left, right in as ref-params.

OK you're right.

> Also, member functions are favoured over lambda callbacks, especially since this is always the same lambda?

Indeed, I originally thought I was going to re-use that lambda, but it was not the case. I'll extract it into an instance method, it'll help reuse. I really wanted to keep the argument count to `_classify_join_conditions` low.  It shouldn't be required to drag everything through: join condition, left and right child of join-node under construction, three vectors, etc.